test/module/irohad/ametsuchi/postgres_query_executor_test.cpp

```diff
@@ -1344,6 +1524,25 @@ namespace iroha {
           result->get()));
     }
 
+    // ------------------------/ tx pagination tests \----------------------- //
+
+    using QueryTxPaginationTestingTypes =
+        ::testing::Types<GetAccountTxPaginationImpl,
+                         GetAccountAssetTxPaginationImpl>;
+    TYPED_TEST_CASE(GetPagedTransactionsExecutorTest,
+                    QueryTxPaginationTestingTypes);
+
+    static const TransactionsPageResponse &getTransactionsPageResponse(
+        const std::unique_ptr<QueryResponse> &query_response) {
+      boost::optional<const TransactionsPageResponse &> tx_page_response;
+      EXPECT_NO_THROW({
```

Comment:
If exception is not thrown, you are guaranteed to have TransactionsPageResponse. Optional is not needed here

Reply:
I need it to store the result. I could also use a pointer, but optional seems more clear.